Challenge1/src/agent.ts

```diff
@@ -1,41 +1,50 @@
-import { Finding, FindingSeverity, FindingType, HandleTransaction, TransactionEvent, ethers } from "forta-agent";
-
-const BOT_DEPLOYMENT_FUNCTION = "function createAgent(uint256 agentId,address ,string metadata,uint256[] chainIds)";
-const BOT_UPDATE_FUNCTION = "function updateAgent(uint256 agentId,string metadata,uint256[] chainIds)";
-const NETHERMIND_ADDRESS = "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8";
-const FORTA_REGISTRY_ADDRESS = "0x61447385B019187daa48e91c55c02AF1F1f3F863";
-
-const provideHandleTransaction = (nethermindAddress: string, fortaRegistryAddress: string): HandleTransaction => {
-  const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent) => {
+import { Finding, FindingSeverity, FindingType, HandleTransaction, TransactionEvent } from "forta-agent";
+import { CREATE_AGENT_ABI, UPDATE_AGENT_ABI, NETHERMIND_ADDRESS, FORTA_REGISTRY_ADDRESS } from "./constants";
+
+export function provideHandleTransaction(
+  createAgentAbi: string,
+  updateAgentAbi: string,
+  nethermindAddress: string,
+  fortaRegistryAddress: string
+): HandleTransaction {
+  return async (txEvent: TransactionEvent) => {
     const findings: Finding[] = [];
 
-    if (txEvent.from !== nethermindAddress) return findings;
-    if (txEvent.to !== fortaRegistryAddress) return findings;
+    if (txEvent.from.toLowerCase() !== nethermindAddress.toLowerCase()) return findings;
+    if (txEvent.to?.toLowerCase() !== fortaRegistryAddress.toLowerCase()) return findings;
 
-    const botEvents = txEvent.filterFunction([BOT_DEPLOYMENT_FUNCTION, BOT_UPDATE_FUNCTION], fortaRegistryAddress);
+    const filteredTxEvents = txEvent.filterFunction([createAgentAbi, updateAgentAbi], fortaRegistryAddress);
 
-    botEvents.forEach((event) => {
+    filteredTxEvents.forEach((event) => {
```

Comment:
```suggestion
    filteredTxEvents.forEach((event: TransactionDescription) => {...
```
In general, we always try to provide the _type_ of variables, arguments, etc. This type would need to be imported from the SDK. The docs state what to expect as the return value type.